.github/workflows/ci.yml

```diff
@@ -295,9 +294,29 @@ jobs:
         if-no-files-found: error
         path: "TestResults/*.trx"
     
+  build-docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install Doc Dependencies
+      working-directory: docs
+      run: | 
+        pip install -r requirements.txt
+    - name: Build Doc
+      working-directory: docs
+      run: |
+        sphinx-build -M html . _build --fail-on-warning
+    - name: Upload HTML Doc Files
+      uses: actions/upload-artifact@v4
+      with:
+        name: html-docs
+        if-no-files-found: error
+        path: "docs/_build/html/**/*"
+    
   release:
     runs-on: ubuntu-latest
-    needs: [build, specs, system-tests-windows, system-tests-linux]
+    needs: [build, specs, system-tests-windows, system-tests-linux, build-docs]
```

Comment:
If the docs artifacts aren't currently used/published, does the release phase really need to wait on it? It _should_ be acting as a sufficient check for PRs, so theoretically it should never fail come release time, but just imagining a scenario where a transient build issue with the docs becomes an annoyance requiring a full rebuild. When the outputs aren't even really "part" of the release!

Reply:
Yeah, makes sense. I was also hesitating about it. I will remove the dep